doc/user/content/integrations/http-api.md

```diff
@@ -0,0 +1,173 @@
+---
+title: "Connect to Materialize via HTTP"
+description: "How to use Materialize via HTTP"
+menu:
+  main:
+    parent: "integrations"
+    weight: 5
+    name: "HTTP API"
+---
+
+You can access Materialize through its "session-less" HTTP API, reachable at
+`https://<MZ host address>/api/sql`.
+
+## Details
+
+### General semantics
+
+The API:
+
+- Requires username/password authentication, just as connecting via `psql`.
+  Materialize provides you the username and password upon setting up your
+  account.
+- Requires you must provide the entirety of your request. The API does not
+  provide session-like semantics, so there is no way to e.g. interactively use
+  transactions.
+- Ceases to process requests upon encountering the first error.
+- Does not support statements whose semantics rely on sessions or whose state is
+  indeterminate at the end of processing, e.g. `DECLARE`, `TAIL`. Trying to use
+  these statements produce an error.
+
+### Transactional semantics
+
+The HTTP API provides two modes with slightly different transactional semantics from one another:
+
+- **Simple**, which mirrors PostgreSQL's [Simple Query][simple-query] protocol.
+    - Supports a single query, but the single query string may contain multiple
+      statements, e.g. `SELECT 1; SELECT 2;`
+    - Treats all statements as in an implicit transaction unless other
+      transaction control is invoked.
+- **Extended**, which mirrors PostgreSQL's [Extended Query][extended-query] protocol.
+    - Supports multiple queries, but only one statement per query string.
+    - Supports parameters.
+    - Eagerly commits DDL (e.g. `CREATE TABLE`) in implicit transactions, but
+      not DML (e.g. `INSERT`).
+
+## Usage
+
+### Endpoint
+
+```
+https://<MZ host address>/api/sql
+```
+
+Accessing the endpoint requires username + password authentication.
```

Comment:
This needs to be defined. A read of the code says we support basic auth with username/password and bearer auth with the full token. We perhaps only need to document basic auth, since full token exchanging is likely not in the interest of most users. https://github.com/MaterializeInc/materialize/blob/ddaa70eacb32ff1c96e3948a81fc68ad2a55898e/src/environmentd/src/http.rs#L666

Reply:
Filed #14959; I'm cordoning off the work on this to simply document the API.